util/file.py

```diff
@@ -830,3 +831,129 @@ def join_paired_fastq(input_fastqs, output_format='fastq', num_n=None):
             }
         rec = SeqRecord(jseq, id=rid, description='', letter_annotations=labbrevs)
         yield rec
+
+
+def repack_tarballs(out_compressed_tarball, input_compressed_tarballs, extract_to_disk_path=None, extract_numeric_owner=False, avoid_disk_roundtrip=True, ignore_zeros=True, pipe_hint=None, threads=None):
+    threads = util.misc.sanitize_thread_count(threads)
+
+    def choose_compressor(filepath, threads=8):
+        return_obj = {}
+        filepath = filepath.lower()
+        if filepath.endswith('.gz'):
+            compressor = 'pigz {threads}'.format(threads="-p "+str(threads) if threads else "").split()
+            return_obj["decompress_cmd"] = compressor + ["-dc"]
+            return_obj["compress_cmd"] = compressor + ["-c"]
+        elif filepath.endswith('.bz2'):
+            #compressor = 'lbzip2 {threads}'.format(threads="-n "+str(threads) if threads else "")
+            compressor = 'lbzip2 {threads}'.format(threads="-n "+str(threads) if threads else "").split()
+            return_obj["decompress_cmd"] = compressor + ["-dc"]
+            return_obj["compress_cmd"] = compressor + ["-c"]
+        elif filepath.endswith('.lz4'):
+            compressor = ['lz4']
+            return_obj["decompress_cmd"] = compressor +["-dc"]
+            return_obj["compress_cmd"] = compressor + ["-c"]
+        elif filepath.endswith('.tar'):
+            compressor = ['cat']
+            return_obj["decompress_cmd"] = compressor
+            return_obj["compress_cmd"] = compressor
+        else:
+            raise IOError("An input file of unknown type was provided: %s" % filepath)
+        return return_obj
+
+    def create_containing_dirs(path):
+        sub_path = os.path.dirname(path)
+        if not os.path.exists(sub_path) and sub_path != path:
+            create_containing_dirs(sub_path)
+        if not os.path.exists(path) and len(path):
+            os.mkdir(path)
+
+    class FileDiverter(object):
```

Comment:
I'm not a huge fan of class inside the function. This function is already very long

Reply:
Yes, I agree, feels ugly but it's also not used beyond here and it relies on the attributes possessing TarInfo attributes so its function is quite internal to this function.